File: javascript/007_10001st_prime.js
function nPrime(n) {
  let count = 0;
  let possiblePrime = 2;
  let latestPrime;

  while (count < n) {
    let isPrime = true;

    for (let i = 2; i < possiblePrime; i++) {
      if (possiblePrime % i === 0) {
        isPrime = false;
        break;
      }
    }

    if(isPrime) {
      latestPrime = possiblePrime;
      count = count + 1;
    }

    possiblePrime = possiblePrime + 1;
  }
  return latestPrime;
}

console.log(nPrime(10001));

File: c/007_10001st_prime.c
#include <stdio.h>
#include <stdbool.h>

int	prime_number(int number)
{
	int possible_prime = 2;
	int latest_prime;
	int count = 0;

	while (count < number)
	{
		bool is_prime = true;
		
		for (int i = 2; i < possible_prime; i++)
		{
			if (possible_prime % i == 0)
			{
				is_prime = false;
				break ;
			}
		}

		if (is_prime)
		{
			latest_prime = possible_prime;
			count = count + 1;
		}

		possible_prime = possible_prime + 1;
	}
	return (latest_prime);
}

int	main(void)
{
	printf("%d\n", prime_number(10001));
	return (0);
}
